internal/attestation/attestation.go

```diff
@@ -38,3 +47,146 @@ type Attestation interface {
 	Statement() []byte
 	Signatures() []signature.EntitySignature
 }
+
+// Extract the payload from a DSSE signature OCI layer
+func PayloadFromSig(sig oci.Signature) (cosign.AttestationPayload, error) {
+	var payload cosign.AttestationPayload
+
+	if sig == nil {
+		return payload, AT001
+	}
+
+	typ, err := sig.MediaType()
+	if err != nil {
+		return payload, AT002.CausedBy(err)
+	}
+
+	if typ != types.DssePayloadType {
+		return payload, AT002.CausedByF("Expecting media type of `%s`, received: `%s`", types.DssePayloadType, typ)
+	}
+
+	reader, err := sig.Uncompressed()
+	if err != nil {
+		return payload, AT002.CausedBy(err)
+	}
+	defer reader.Close()
+
+	payloadBytes, err := io.ReadAll(reader)
+	if err != nil {
+		return payload, AT002.CausedBy(err)
+	}
+
+	err = json.Unmarshal(payloadBytes, &payload)
+	if err != nil {
+		return payload, AT002.CausedBy(err)
+	}
```

Comment:
If we wanted to conserve a bit of memory (*citation needed) this could go through a Decoder, something like:
```suggestion
	if err := json.NewDecoder(reader).Decode(&payload), err != nil {
		return payload, AT002.CausedBy(err)
	}
```

Reply:
It works, but I'll put it in a separate PR.